docs/en/04_Changelogs/3.2.0.md

```diff
@@ -221,121 +490,214 @@ As a result of this upgrade there are now very few cases where `Convert::raw2sql
+
+The place where legacy code would almost certainly fail is any code that calls
+DataList::sql`, `DataQuery::sql`, `SQLQuery::sql` or `SQLQuery::getJoins()`, as the api requires that user
+code passes in an argument here to retrieve SQL parameters by value.
+
+User code that assumes parameterless queries will likely fail, and need to be
+updated to handle this case properly.
+
+Before:
+
+
+	:::php
+	<?php
+
+	// Generate query
+	$argument = 'whatever';
+	$query = SQLQuery::create()
+		->setFrom('"SiteTree"')
+		->setWhere(array("\"SiteTree\".\"Title\" LIKE '" . Convert::raw2sql($argument) . "'"));
+
+	// Inspect elements of the query
+	$sql = $query->sql();
+	$sql = preg_replace('/LIKE \'(.+)\'/', 'LIKE \'%${1}%\'', $sql); // Adds %% around the argument
+
+	// Pass new query to database connector
+	DB::query($sql);
+
+
+After:
+
+
+	:::php
+	<?php
+
+	// Generate query
+	$argument = 'whatever';
+	$query = SQLQuery::create()
+		->setFrom('"SiteTree"')
+		->setWhere(array('"SiteTree"."Title" LIKE ?' => $argument));
+
+	// Inspect elements of the query
+	$sql = $query->sql($parameters);
+	foreach($parameters as $key => $value) {
+		// Adds %% around arguments
+		$parameters[$key] = "%{$value}%";
+	}
+
+	// Pass new query to database connector
+	// Note that DB::query($sql) would fail, as it would contain ? with missing parameters
+	DB::prepared_query($sql, $parameters);
+
+
+Also note that the parameters may not be a single level array, as certain values
+may be forced to be cast as a certain type (where supported by the current API).
+
+E.g.
+
+
+	:::php
+	<?php
 
-		:::php
-		<?php
+	$parameters = array(
+		10,
+		array('value' => 0, 'type' => 'boolean') // May also contain other database API specific options
+	)
+	DB::prepared_query('DELETE FROM "MyObject" WHERE ParentID = ? OR IsValid = ?', $parameters);
 
-		// Generate query
-		$argument = 'whatever';
-		$query = SQLQuery::create()
-			->setFrom('"SiteTree"')
-			->setWhere(array('"SiteTree"."Title" LIKE ?' => $argument));
 
-		// Inspect elements of the query
-		$sql = $query->sql($parameters);
-		foreach($parameters as $key => $value) {
-			// Adds %% around arguments
-			$parameters[$key] = "%{$value}%";
+#### 4. Interaction with `SQLQuery::getWhere()` method
+
+As all where conditions are now parameterised, the format of the results returned by `SQLQuery::getWhere()`
+will not always equate with that in FrameWork 3.1. Once this would be a list of strings, what will
+now be returned is a list of conditions, each of which is an associative array mapping the
+condition string to a list of parameters provided.
+
+Before:
+
+
+	:::php
+	<?php
+
+	// Increment value of a single condition
+	$conditions = $query->getWhere();
+	$new = array();
+	foreach($conditions as $condition) {
+		if(preg_match('/\"Count\" = (?<count>\d+)/', $condition, $matches)) {
+			$condition = '"Count" = '.($matches['count'] + 3);
```

Comment:
I'm terrible at coming up with real world examples.

I'm using +3 because that's whatever random logic this method is trying to do.